src/extensions/foot/footext_hooks.cpp

```diff
@@ -35,9 +35,94 @@
 #include "asserthandler.h"
 #include "debughandler.h"
 
+#include "infantry.h"
+#include "unit.h"
+
 #include "hooker.h"
 #include "hooker_macros.h"
+#include "tibsun_defines.h"
+
+/*
+* The Foot object initializes the passenger.
+* 
+* Injection objective function: FootClass::Unlimbo
+* 
+* @author: RossinCarlinx
+*/
+static void __stdcall FootClass_Initialize_Passengers(FootClass* this_ptr, TechnoTypeClassExtension* technotypeext) {
+    auto const Count = technotypeext->InitPassengers.Count();
+    // Check if the initial number of passenger types is greater than 0.
+    if (Count > 0) {
+
+        auto const house = this_ptr->House;
+        // Iterates through objects in InitPassenger, adding them if they are infantry or units.
+        for (int i = 0;
+            i < Count;
+            i++) {
+            auto const technotype = technotypeext->InitPassengers[i];
+            auto const rtti = technotype->Kind_Of();
+
+            if (rtti != RTTI_INFANTRYTYPE &&
+                rtti != RTTI_UNITTYPE) {
+                continue;
+            }
+
+            auto num = (technotypeext->InitPassengerNums.Count() > i) ?
+                technotypeext->InitPassengerNums[i] :
+                1;
+
+            // A loop that creates objects and puts them into the position of passengers.
+            do {
+                auto const object = technotype->Create_One_Of(house);
+                auto const passenger = object->As_Foot();
+                if (!passenger) {
+                    Fatal("*Error* Object creation failed.\n\tWhere the error occurred: \n\FootClass_Initialize_Passengers\n");
+                }
+
+                // Prevent the generation of initial occupants indefinitely.
+                if (passenger->Class_Of() == this_ptr->Class_Of()) {
+                    Extension::Fetch<TechnoClassExtension>(passenger)->Initialized = true;
+                }
+
+                passenger->Set_Coord(this_ptr->Get_Coord());
+                passenger->Limbo();
+                this_ptr->Cargo.Attach(passenger);
+#ifndef NDEBUG
+                Vinifera_Output_Debug_String("this_ptr->Cargo.Attach();");
+#endif
+            } while (--num > 0);
+        }
+    }
+}
+DECLARE_PATCH(_FootClass_Initialize_Passengers_Patch) {
+    GET_REGISTER_STATIC(FootClass*, this_ptr, edi);
+    static TechnoClassExtension* technoclassext;
+    static TechnoTypeClassExtension* technotypeext;
+
+    technoclassext = Extension::Fetch<TechnoClassExtension>(this_ptr);
+    technotypeext = Extension::Fetch<TechnoTypeClassExtension>(this_ptr->Techno_Type_Class());
+    
+    // Original location code.
+    if (!this_ptr->Locomotion) {
+        goto error_return;
+    }
+
+    // If it has already been initialized, it is automatically skipped.
+    if (technoclassext->Initialized) {
+        goto normal;
+    }
+
+    technoclassext->Initialized = true;
 
+    // Perform initialization of passengers.
+    FootClass_Initialize_Passengers(this_ptr, technotypeext);
```

Comment:
See the first comment of this review. You can perform a extension class lookup on `this_ptr` and the function can be moved to FootClassExtension.